### PR TITLE
bitmap: use calloc instead of malloc/memset

### PR DESCRIPTION
**Pull request by rcombs**
This can improve perf significantly with large bitmaps

### CODE REVIEW

**Comment by grigorig** [comment]
Can you provide numbers for the performance improvements?

**Comment by rcombs** [comment]
I'm seeing improvements in the range of 5% on the particularly heavy (on allocation size) lines in [this sample](http://puu.sh/pE9Uc/2315b54f8b.ass), so I suppose it's "significant" in the statistical sense. It ended up appearing more dramatic during playback on my machine, since we were already very close to realtime, so a minor performance improvement ended up getting the per-frame render time down low enough that I went from 12 dropped frames to 1 pretty reliably.

The point is to avoid the `memset()` call in cases where `calloc()` is allocating (kernel-zeroed) pages directly. `calloc()` acts like `malloc()` + `memset()` in other cases, so it should be no slower than the current code there.

File: libass/ass_utils.c [hunk 1]
```diff
@@ -95,6 +95,22 @@ void *ass_aligned_alloc(size_t alignment, size_t size)
...
+    if (!allocation)
+        return NULL;
+    char *ptr = allocation + sizeof(void *);
+    unsigned int misalign = (uintptr_t)ptr & (alignment - 1);
+    if (misalign)
+        ptr += alignment - misalign;
+    *((void **)ptr - 1) = allocation;
+    return ptr;
+}
+
```
Comment (wm4) [inline libass/ass_utils.c]: Seems good overall. I wonder if you could reduce code duplication here by a bit by moving the aligning to a common function?

**Comment by grigorig** [comment]
Sounds good, LGTM if you address the code duplication.

**Comment by MrSmile** [comment]
For consistency, I think `ass_aligned_alloc()` and `ass_aligned_calloc()` should be combined (with similar `bool zero` parameter). Also `alloc_bitmap()` and `alloc_bitmap_raw()`.

**Comment by rcombs** [comment]
Yeah, that's what I'll end up doing. This is basically how malloc and calloc work internally (on OS X) as well.

**Comment by grigorig** [comment]
LGTM.

**Comment by grigorig** [comment]
Merged.

**Pull request closed without merging**